pkg/build/pipelines/git-checkout.yaml

```diff
@@ -1,148 +1,181 @@
-          # exists both a tag and a branch by that name.
-          # since a tag was given, we want the tag.
-          vr git fetch $remote ${depthflag:+"$depthflag"} --no-tags \
-              "+refs/tags/$tag:refs/$remote/tags/$tag"
-          vr git checkout "$remote/tags/$tag"
-
-          foundcommit=$(git rev-parse --verify HEAD)
-          if [ -z "$expcommit" ] || [ "$expcommit" = "$foundcommit" ]; then
-              msg "tag $tag is $foundcommit"
-              return 0
-          fi
-
-          # If it's a tag, then it could be a lightweight or annotated tag.
-          # Lightweight tags point directly to the commit and do not have
-          # any messages, signatures, or other data.  Annotated tags point
-          # to its own git object containing the tag data, with a reference
-          # to the underlying commit.  We expect most tags to be using
-          # annotated tags.
-          tagobj=$(git rev-parse --verify --end-of-options \
-              "refs/$remote/tags/$tag")
-          if [ "$expcommit" != "$tagobj" ]; then
-              [ "$tagobj" != "$expcommit" ] &&
-                  msg "tag object hash was $tagobj"
-              fail "Expected commit $expcommit for $tag, found $foundcommit"
-          fi
-
-          msg "Warning: The provided expected-commit ($expcommit)"
-          msg "was the hash of the annotated tag object for $tag."
-          msg "Update to set expected-commit to $foundcommit"
-
-          return 0
-      }
+
+        msg() { echo "[git checkout]" "$@" 1>&2; }
+        fail() { msg FAIL "$@"; exit 1; }
+        vr() { msg "execute:" "$@"; "$@"; }
+
+        process_cherry_picks() {
+            local cpicksf="$1" oifs="" count=0
+            local fetched_branches=""
+            if [ -z "$cpicksf" ]; then
+                return 0
+            fi
+            if [ ! -f "$cpicksf" ]; then
+                msg "cherry picks input '$cpicksf' is not a file"
+                return 1
+            fi
+
+            local line="" branch="" hash=""
+            while read line; do
+                # drop anything after #
+                line=${line%%#*}
+                [ -z "$line" ] && continue
+                case "$line" in
+                    */*) branch=${line%%/*}; hash=${line#*/};;
+                    *) hash=${line};;
+                esac
+                if [ -n "$branch" ]; then
+                    case " $fetched_branches " in
+                        *) vr git fetch origin $branch:$branch || {
+                            msg "failed to fetch branch $branch"
+                            return 1
+                            }
+                            local fetched_branches="$fetched_branches $branch";;
+                    esac
+                fi
+                vr env GIT_AUTHOR_NAME="Melange Build" GIT_AUTHOR_EMAIL="melange-build@cgr.dev" GIT_COMMITTER_DATE="$SOURCE_EPOCH_DATE" git cherry-pick -x "$hash" || {
```

Comment:
make this line shorter. 

GIT_AUTHOR_NAME=.... \
    GIT_COMMITER_DATE=... \
    git cherry-pick -x